src/locationinfo.jl

```diff
@@ -35,9 +35,10 @@ end
 function locinfo(frame::Frame)
     if frame.framecode.scope isa Method
         meth = frame.framecode.scope
-        file, def_line = JuliaInterpreter.whereis(meth)
-        _, current_line = JuliaInterpreter.whereis(frame)
-        return loc_for_fname(file, current_line, def_line)
+        def_file, def_line = JuliaInterpreter.whereis(meth)
+        current_file, current_line = JuliaInterpreter.whereis(frame)
+        def_line > current_line && (def_line = 1) # This can happen in e.g. macro expansions
```

Comment:
Is this just to prevent `BoundsError`? Otherwise it doesn't seem particularly reliable.

Reply:
Bascially `def_line` provides a "roof" where we do not print source code above it. Because if we are in `sin` we do not want to print code above the `sin` definition.

However, if we are in another file such a roof doesnt make sense. Setting this to `1` just mean that we will print the maximum number of source lines.

Reply:
But yeah, perhaps this is the wrong place to do this since it isn't really location info anymore, just data for the source code printer to do its job as good as it can.

Reply:
Thanks for the explanation. I'm fine with whatever you choose. Perhaps add a code comment explaining these issues?

Reply:
Actually this should be `= 0` since the commend for `def_line` says 

```
    # The line at which the current context starts, 0 if unknown
```

Reply:
Pushed an update